### PR TITLE
Remote player identification

### PR DESCRIPTION
**Pull request by friedow**

### CODE REVIEW

File: src/tetris/field/remoteField.ts [hunk 1]
```diff
@@ -75,5 +100,41 @@ export default class RemoteField {
...
 	}
+
+	private _setupText() {
+		this._name = this._scene.add.text(this._drawOffset.x, this._drawOffset.y, " ", config.ui.fonts.scoreboard.font);
+		this._score = this._scene.add.text(this._drawOffset.x, this._drawOffset.y, " ", config.ui.fonts.scoreboard.font);
+		this._adjustTextX();
+		this._adjustTextY();
+	}
+
+	private _updateSprites(serializedBlockState) {
```
Comment (msoechting) [inline src/tetris/field/remoteField.ts]: Data type missing (`Array<Array<SerializedBlockState>>`).
Also the parameter name could be more precise (e.g. serializedFieldState)

File: src/tetris/field/remoteField.ts [hunk 2]
```diff
@@ -2,24 +2,42 @@
...
-			for (let x = 0; x < this._width; x++) {
-				if (serializedBlocks[y][x]) {
-					this._blockRows[y][x].sprite.setVisible(true);
-					this._blockRows[y][x].sprite.setFrame(serializedBlocks[y][x].spriteFrameName);
-				} else {
-					this._blockRows[y][x].sprite.setVisible(false);
-				}
-			}
-		}
+	public update(player: MatchPlayer) {
```
Comment (msoechting) [inline src/tetris/field/remoteField.ts]: I don't think this is a MatchPlayer; see the differences between MatchPlayer and Serialized(Match)Player in the interfaces

**Comment by msoechting** [comment]
grafik
I had 2 players in the match, first one called "12", second one "asdf", and in the game screen of the first player, only his own name seemed to be displayed correctly. The other three text fields seem to be black rectangles.

File: src/tetris/field/remoteField.ts (hunk 2 above)
Comment (friedow) [reply src/tetris/field/remoteField.ts]: I dont see any serialized players. Where exactly is the changed interface?

**Comment by friedow** [comment]
Could not reproduce your issue. Could you try again, please?
image

**Comment by msoechting** [comment]
grafik
Just as I suspected: black rectangles. I also get this error thrown on my console every frame:
`[.Offscreen-For-WebGL-000001AC9A79F110]RENDER WARNING: texture bound to texture unit 0 is not renderable. It maybe non-power-of-2 and have incompatible texture filtering.`
This error does not occur on the developer branch.

File: src/tetris/field/remoteField.ts (hunk 2 above)
Comment (msoechting) [reply src/tetris/field/remoteField.ts]: Apparently, it's only [in the server code](https://github.com/TeamTetris/tetris-server/blob/1b0be662547914f9d426edbfd147aa377f4ec1b3/src/player/serializedMatchPlayer.ts).

File: src/tetris/field/remoteField.ts [hunk 3]
```diff
@@ -2,27 +2,45 @@
 
 import Block from 'tetris/brick/block';
 import Vector2 = Phaser.Math.Vector2;
+import MatchPlayer, { BlockState } from 'tetris/interfaces/MatchPlayer';
 import config from 'tetris/config';
 
 export default class RemoteField {
 	//region public members
+	get height(): number {
```
Comment (Johannes-Schneider) [inline src/tetris/field/remoteField.ts]: All of these should be public, right?

File: src/tetris/field/remoteField.ts [hunk 4]
```diff
@@ -75,5 +100,41 @@ export default class RemoteField {
 			}
 		}
 	}
+
+	private _setupText(): void {
```
Comment (Johannes-Schneider) [inline src/tetris/field/remoteField.ts]: `_setupText` is very generic... perhaps you can find a better name for this method.

**Comment by friedow** [comment]
It seems like the rendering error only occurs on windows. How can we deal with that?

**Comment by msoechting** [comment]
We can avoid our font hack (and possibly solve the issue at hand) by using [BitmapText](https://photonstorm.github.io/phaser3-docs/Phaser.GameObjects.BitmapText.html). Usage looks like shaders: `this.load.bitmapFont(..); this.add.bitmapText(16, 0, 'font-name', 'SCORE: 0', 32);`

File: src/tetris/ui/scoreWidget.ts [hunk 5]
```diff
@@ -10,28 +10,28 @@ export default class ScoreWidget {
 	//endregion
 
 	//region public methods
-	get height(): number {
+	public get height(): number {
```
Comment (Johannes-Schneider) [inline src/tetris/ui/scoreWidget.ts]: These should be `public members`

File: src/tetris/ui/textButton.ts [hunk 6]
```diff
@@ -11,28 +11,28 @@ export default class TextButton {
 	//endregion
 
 	//region public methods
-	get height(): number {
+	public get height(): number {
```
Comment (Johannes-Schneider) [inline src/tetris/ui/textButton.ts]: These should be `public members`

File: src/tetris/ui/countdownWidget.ts [hunk 7]
```diff
@@ -11,28 +11,28 @@ export default class CountdownWidget {
 	//endregion
 
 	//region public methods
-	get height(): number {
+	public get height(): number {
```
Comment (Johannes-Schneider) [inline src/tetris/ui/countdownWidget.ts]: These should be `public members`

**Pull request merged**